validator/telegram/telegram.go

```diff
@@ -0,0 +1,173 @@
+package telegram
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/nextdotid/proof_server/config"
+	"github.com/nextdotid/proof_server/types"
+	"github.com/nextdotid/proof_server/util"
+	mycrypto "github.com/nextdotid/proof_server/util/crypto"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/xerrors"
+
+	"github.com/gotd/td/telegram"
+	"github.com/gotd/td/tg"
+	"github.com/nextdotid/proof_server/validator"
+)
+
+type Telegram struct {
+	*validator.Base
+}
+
+const (
+	MATCH_TEMPLATE = "^Sig: (.*)$"
+)
+
+var (
+	client      *telegram.Client
+	l           = logrus.WithFields(logrus.Fields{"module": "validator", "validator": "telegram"})
+	re          = regexp.MustCompile(MATCH_TEMPLATE)
+	POST_STRUCT = map[string]string{
+		"default": "🎭 Verifying my Telegram ID @%s for @NextDotID.\nSig: %%SIG_BASE64%%\n\nPowered by Next.ID - Connect All Digital Identities.\n",
+		"en_US":   "🎭 Verifying my Telegram ID @%s for @NextDotID.\nSig: %%SIG_BASE64%%\n\nPowered by Next.ID - Connect All Digital Identities.\n",
+		"zh_CN":   "🎭 正在通过 @NextDotID 验证我的 Telegram 帐号 @%s 。\nSig: %%SIG_BASE64%%\n\n由 Next.ID 支持 - 连接全域数字身份。\n",
+	}
+)
+
+func Init() {
+	initClient()
+	if validator.PlatformFactories == nil {
+		validator.PlatformFactories = make(map[types.Platform]func(*validator.Base) validator.IValidator)
+	}
+
+	validator.PlatformFactories[types.Platforms.Telegram] = func(base *validator.Base) validator.IValidator {
+		telg := Telegram{base}
+		return &telg
+	}
+}
+
+func (telegram *Telegram) GeneratePostPayload() (post map[string]string) {
+	post = make(map[string]string, 0)
+	for lang_code, template := range POST_STRUCT {
+		post[lang_code] = fmt.Sprintf(template, telegram.Identity)
+	}
+
+	return post
+}
+
+func (telegram *Telegram) GenerateSignPayload() (payload string) {
+	telegram.Identity = strings.ToLower(telegram.Identity)
+	payloadStruct := validator.H{
+		"action":     string(telegram.Action),
+		"identity":   telegram.Identity,
```

Comment:
OK for now, but do you have a way to get long-digit telegram user ID instead of a modifiable `username` here before generating `sign_payload` ?
We want to make this validation lasts as long as it can. User can changes their `@my_username` after the proof post, but `LONG_DIGIT_USER_ID` will remain the same.

Reply:
Take `validator/steam` as a sample if you want.

Reply:
Definitely right, thanks for the heads up!